primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java

```diff
@@ -152,13 +156,19 @@ public static void goTo(String partialUrl) {
      * @return the URL of the page
      */
     public static String getUrl(AbstractPrimePage page) {
-        DeploymentAdapter deploymentAdapter = ConfigProvider.getInstance().getDeploymentAdapter();
-
         String baseLocation = page.getBaseLocation();
-        if (deploymentAdapter != null) {
+        if (baseLocation == null) {
             baseLocation = getBaseUrl();
         }
-
+        if (baseLocation == null) {
+            StringBuilder messageBuilder = new StringBuilder("Cannot determine base url. Please either configure ").append(DEPLOYMENT_BASEURL).append(" or ");
```

Comment:
is this required? in case its null, the RuntimeException will be already thrown in #getBaseUrl?

Reply:
This is not fully duplication: page's class has method getBaseLocation with same functionality as getBaseUrl. This possibility (to configure baseUrl for each page) could be also noticed in error's message.